tests/tests_integration/test_api/test_workflows.py

```diff
@@ -184,10 +184,12 @@ def add_multiply_workflow(
             ],
         ),
     )
-    # Delete existing workflow and versions
-    cognite_client.workflows.delete(workflow_id, ignore_unknown_ids=True)
-    yield cognite_client.workflows.versions.upsert(version)
-    cognite_client.workflows.delete(workflow_id, ignore_unknown_ids=True)
+
+    retrieved = cognite_client.workflows.versions.retrieve(version.workflow_external_id, version.version)
+    if retrieved is not None:
+        return retrieved
+    else:
+        return cognite_client.workflows.versions.upsert(version)
```

Comment:
Avoid deleting the workflow every time, and instead reuse the same workflow. The problem seems to be that the workflow is sometimes takes longer to create, and then listing executions fails.